// angular-animations/esm2015/flippers/flip-out-x.animation.js
/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,uselessCode} checked by tsc
 */
import { animate, animateChild, animation, group, keyframes, query, style, transition, trigger, useAnimation } from '@angular/animations';
/**
 * @record
 */
export function IFlipOutXAnimationOptions() { }
if (false) {
    /**
     * Angle - number of degrees at which end animation.
     *
     * Default 90
     * @type {?|undefined}
     */
    IFlipOutXAnimationOptions.prototype.degrees;
}
/** @type {?} */
const flipOutX = animation([
    animate('{{duration}}ms {{delay}}ms', keyframes([
        style({ transform: 'perspective(400px)', opacity: 1, easing: 'ease', offset: 0 }),
        style({ transform: 'perspective(400px) rotate3d(1, 0, 0, -20deg)', opacity: 1, easing: 'ease', offset: 0.3 }),
        style({ transform: 'perspective(400px) rotate3d(1, 0, 0, {{degrees}}deg)', opacity: 0, easing: 'ease', offset: 1 })
    ]))
]);
/** @type {?} */
const DEFAULT_DURATION = 750;
/**
 * @param {?=} options
 * @return {?}
 */
export function flipOutXAnimation(options) {
    return trigger((options && options.anchor) || 'flipOutX', [
        transition('0 => 1', [
            ...(options && options.animateChildren === 'before' ? [query('@*', animateChild(), { optional: true })] : []),
            group([
                style({ 'backface-visibility': 'visible' }),
                useAnimation(flipOutX),
                ...(!options || !options.animateChildren || options.animateChildren === 'together'
                    ? [query('@*', animateChild(), { optional: true })]
                    : [])
            ]),
            ...(options && options.animateChildren === 'after' ? [query('@*', animateChild(), { optional: true })] : [])
        ], {
            params: {
                delay: (options && options.delay) || 0,
                duration: (options && options.duration) || DEFAULT_DURATION,
                degrees: (options && options.degrees) || 90
            }
        })
    ]);
}
/**
 * @param {?=} options
 * @return {?}
 */
export function flipOutXOnLeaveAnimation(options) {
    return trigger((options && options.anchor) || 'flipOutXOnLeave', [
        transition(':leave', [
            ...(options && options.animateChildren === 'before' ? [query('@*', animateChild(), { optional: true })] : []),
            group([
                style({ 'backface-visibility': 'visible' }),
                useAnimation(flipOutX),
                ...(!options || !options.animateChildren || options.animateChildren === 'together'
                    ? [query('@*', animateChild(), { optional: true })]
                    : [])
            ]),
            ...(options && options.animateChildren === 'after' ? [query('@*', animateChild(), { optional: true })] : [])
        ], {
            params: {
                delay: (options && options.delay) || 0,
                duration: (options && options.duration) || DEFAULT_DURATION,
                degrees: (options && options.degrees) || 90
            }
        })
    ]);
}
